easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1379,41 +1407,59 @@ def copy_easyconfigs(paths, target_dir):
-                name = file_info['ecs'][-1].name
-                ec_filename = '%s-%s.eb' % (name, det_full_ec_version(file_info['ecs'][-1]))
+            target_path = det_easyconfig_location(path, target_dir, soft_name, ec_filename)
+            file_info['new'].append(not os.path.exists(target_path))
+
+            try:
+                mkdir(os.path.dirname(target_path), parents=True)
+                shutil.copy2(path, target_path)
+                _log.info("%s copied to %s", path, target_path)
+            except OSError as err:
+                raise EasyBuildError("Failed to copy %s to %s: %s", path, target_path, err)
 
-                letter = name.lower()[0]
-                if letter not in a_to_z:
-                    raise EasyBuildError("Don't know which letter subdir to use for %s", name)
+            file_info['paths_in_repo'].append(target_path)
 
-                target_path = os.path.join(subdir, letter, name, ec_filename)
-                _log.debug("Target path for %s: %s", path, target_path)
+        else:
+            raise EasyBuildError("Multiple EasyConfig instances obtained from easyconfig file %s", path)
 
-                full_target_path = os.path.join(target_dir, target_path)
-                try:
-                    file_info['new'].append(not os.path.exists(full_target_path))
+    return file_info
 
-                    mkdir(os.path.dirname(full_target_path), parents=True)
-                    shutil.copy2(path, full_target_path)
-                    _log.info("%s copied to %s", path, full_target_path)
-                except OSError as err:
-                    raise EasyBuildError("Failed to copy %s to %s: %s", path, target_path, err)
 
-                file_info['paths_in_repo'].append(target_path)
-            else:
-                raise EasyBuildError("Multiple EasyConfig instances obtained from easyconfig file %s", path)
-    else:
-        raise EasyBuildError("Subdirectory %s not found in %s", subdir, target_dir)
+def copy_patch_files(patch_specs, target_dir):
+    """
+    Copy patch files to specified directory, in the 'right' location according to the software name they relate to.
 
-    return file_info
+    @param patch_specs: list of tuples with patch file location and name of software they are for
+    @param target_dir: target directory
+    """
+    patched_files = {
+        'files': [],
+        'paths_in_repo': [],
+        'new': [],
+    }
+    for patch_path, soft_name in patch_specs:
+        target_path = det_easyconfig_location(patch_path, target_dir, soft_name, os.path.basename(patch_path))
+        patched_files['new'].append(not os.path.exists(target_path))
+
+        try:
+            mkdir(os.path.dirname(target_path), parents=True)
+            shutil.copy2(patch_path, target_path)
+            _log.info("%s copied to %s", patch_path, target_path)
+
+        except OSError as err:
+            raise EasyBuildError("Failed to copy %s to %s: %s", patch_path, target_path, err)
```

Comment:
this whole block is copy-pasted, let's make a `copy_file` function in `filetools` for this?

I'm sure it can be used in other places as well (but that's beyond the scope of this PR)...